### PR TITLE
Add garage-check utility

### PR DESCRIPTION
**Pull request by OYTIS**
TreehubServer is responsible for authentication how it's done on the build machine (i.e. without a registered device), so I've made it do the job for accessing TUF repo as well. Which is of course a questionable solution.

### CODE REVIEW

File: src/sota_tools/garage_check.cc [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    LOG_FATAL << "Error initializing curl";
+    return EXIT_FAILURE;
+  }
+  curl_easy_setopt(curl_handle, CURLOPT_VERBOSE, get_curlopt_verbose());
+  curl_easy_setopt(curl_handle, CURLOPT_NOBODY, 1);  // HEAD
+
+  treehub.InjectIntoCurl("objects/" + ref.substr(0, 2) + "/" + ref.substr(2) + ".commit", curl_handle);
+
+  CURLcode result = curl_easy_perform(curl_handle);
+  curl_easy_cleanup(curl_handle);
```
Comment (cajun-rat) [inline src/sota_tools/garage_check.cc]: I think this should be after curl_easy_getinfo() a few lines down?

File: src/sota_tools/garage_check.cc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  }
+
+  // check if the ref is present on treehub
+  CURL *curl_handle = curl_easy_init();
+  if (!curl_handle) {
+    LOG_FATAL << "Error initializing curl";
+    return EXIT_FAILURE;
+  }
+  curl_easy_setopt(curl_handle, CURLOPT_VERBOSE, get_curlopt_verbose());
+  curl_easy_setopt(curl_handle, CURLOPT_NOBODY, 1);  // HEAD
```
Comment (cajun-rat) [inline src/sota_tools/garage_check.cc]: Do we need a `1L` to force a long here?

File: src/sota_tools/garage_check.cc [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    LOG_FATAL << "OSTree commit " << ref << " is missing in treehub";
+    return EXIT_FAILURE;
+  } else if (http_code != 200) {
+    LOG_FATAL << "Error " << http_code << " getting commit " << ref << " from treehub";
+    return EXIT_FAILURE;
+  }
+  LOG_INFO << "OSTree commit " << ref << " is found on treehub";
+
+  // check if the ref is present in targets.json
+  curl_handle = curl_easy_init();
```
Comment (cajun-rat) [inline src/sota_tools/garage_check.cc]: ...actually I think we can reuse the curl_handle instead

**Comment by cajun-rat** [comment]
A couple of minor issues, but otherwise good to go

File: src/sota_tools/garage_check.cc (hunk 2 above)
Comment (OYTIS) [reply src/sota_tools/garage_check.cc]: Yes, it's a long, better to specify explicilty.

File: src/sota_tools/garage_check.cc (hunk 1 above)
Comment (OYTIS) [reply src/sota_tools/garage_check.cc]: Makes sense, yes.

File: src/sota_tools/garage_check.cc (hunk 3 above)
Comment (OYTIS) [reply src/sota_tools/garage_check.cc]: I'll test that.

**Pull request merged**